src/main/java/org/jenkinsci/plugins/GithubSecurityRealm.java

```diff
@@ -659,7 +659,7 @@ public UserDetails loadUserByUsername(String username)
 
         Authentication token = SecurityContextHolder.getContext().getAuthentication();
 
-        if (token == null) {
+        if (token == null || username != token.getPrincipal()) {
```

Comment:
```suggestion
        if (token == null || !username.equals(token.getPrincipal())) {
```

String reference comparison could be problematic.